src/main/java/Duke.java

```diff
@@ -1,10 +1,431 @@
+        return resultList;
+    }
+
+    
+    
+    
+    
+    
+    public static void printGreet(){
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke_HTT. \n What can I do for you?");
+    }
+
+    protected static String getTaskMark(String[] inLine) {
+        int inputTaskIndex = Integer.parseInt(inLine[1]);
+        String taskType = tasks.get(inputTaskIndex - 1).getTaskType();
+        String taskMark = "";
+        switch (taskType) {
+        case "ToDo":
+            taskMark = "[T]";
+            break;
+
+        case "Deadline":
+            taskMark = "[D]";
+            break;
+
+        case "Event":
+            taskMark = "[E]";
+            break;
+
+        default:
+            break;
+        }
+        return taskMark;
+    }
+
+    public static void processInput(){   
+        Scanner scan = new Scanner ( System.in ); 
+        String inData = scan.nextLine(); // user line of input
+        String[] inLine = inData.split(" "); // code process line of input into an array
+        
+        boolean isFirstLine = false; 
+
+        while (!inData.equals("bye")){
+            if (isFirstLine) { // kick start the loop if first time
+                inData = scan.nextLine();
+                inLine = inData.split(" ");
+            } else {
+                isFirstLine = true;
+            }
+
+            int inputTaskIndex = Integer.parseInt(inLine[1]);
+            int taskIndex = inputTaskIndex - 1;
+            switch (inLine[0]){
+            case "list":
+                printList(tasks);
+                break;
+
+            case "mark":
+            try {
+                if (inputTaskIndex > tasks.size() || inputTaskIndex < 1) {
+                    throw new InvalidListIndexException();
+                }
+
+                tasks.get(taskIndex).setTaskDone(true);
+                
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
+
+                System.out.println("Nice! I've marked this task as done: \n" + taskMark + doneMark +
+                        tasks.get(taskIndex).getTaskTitle());
+
+                String newEntry = parseTaskToString(tasks);
+                writeDisk(newEntry);
+            } catch (InvalidListIndexException | NumberFormatException e) {
+                System.out.println("Error - List index given invalid. Please check again.");
+            } catch (IOException e) {
+                System.out.println("Error when saving file: " + e.getMessage());
+            }
+            break;
```

Comment:
You should consider indenting this block of code one level further, as stated in the coding standard, and to maintian consistency across your code.